akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/PersistentActorImpl.scala

```diff
@@ -107,21 +103,31 @@ import akka.persistence.journal.Tagged
       val eventToPersist = if (tags.isEmpty) event else Tagged(event, tags)
       persist(eventToPersist) { _ ⇒
         sideEffects.foreach(applySideEffect)
+        if (shouldSnapshot(state, event))
+          saveSnapshot(state)
       }
     case PersistAll(events) ⇒
       if (events.nonEmpty) {
         // apply the event before persist so that validation exception is handled before persisting
         // the invalid event, in case such validation is implemented in the event handler.
         // also, ensure that there is an event handler for each single event
         var count = events.size
-        state = events.foldLeft(state)(applyEvent)
+        val (newState, shouldSnapshotAfterPersist) = events.foldLeft((state, false)) {
+          case ((currentState, snapshot), event) ⇒
+            (applyEvent(currentState, event), snapshot || shouldSnapshot(currentState, event))
```

Comment:
the `lastSequenceNr` is not updated yet so you have to +1 for each event here,

might want to add a test for the auto-snapshotting when the seqNr is matching in the middle of a persistAll batch

Reply:
Added the test